app/src/main/java/org/dhis2/data/forms/dataentry/FormView.kt

```diff
@@ -191,9 +247,63 @@ class FormView private constructor(
         else -> false
     }
 
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        if (resultCode == Activity.RESULT_OK &&
+            requestCode == Constants.RQ_MAP_LOCATION_VIEW && data?.extras != null
+        ) {
+            val locationType =
+                FeatureType.valueOf(data.getStringExtra(MapSelectorActivity.LOCATION_TYPE_EXTRA))
+            val dataExtra = data.getStringExtra(MapSelectorActivity.DATA_EXTRA)
+            val geometry: Geometry = when (locationType) {
+                FeatureType.POINT -> {
+                    val type = object : TypeToken<List<Double?>?>() {}.type
+                    GeometryHelper.createPointGeometry(
+                        Gson().fromJson(dataExtra, type)
+                    )
+                }
+                FeatureType.POLYGON -> {
+                    val type = object : TypeToken<List<List<List<Double?>?>?>?>() {}.type
+                    GeometryHelper.createPolygonGeometry(
+                        Gson().fromJson(dataExtra, type)
+                    )
+                }
+                else -> {
+                    val type = object : TypeToken<List<List<List<List<Double?>?>?>?>?>() {}.type
+                    GeometryHelper.createMultiPolygonGeometry(
+                        Gson().fromJson(dataExtra, type)
+                    )
+                }
+            }
```

Comment:
why not having it in the viewModel or a mapper and be able to test it?

Reply:
I'll move it to a helper class then, so we can reuse it